### PR TITLE
[Server] Upgrade to `v1beta1` schema.

### PR DESCRIPTION
**Pull request by MUzairS15**
**Description**

This PR fixes #

**Notes for Reviewers**

**[Signed commits](../CONTRIBUTING.md#signing-off-on-commits-developer-certificate-of-origin)**
- [ ] Yes, I signed my commits.

### CODE REVIEW

**Comment by leecalcote** [comment]
A month old at this point…. Merge conflicts.

**Comment by MUzairS15** [comment]
// @leecalcote @Yashsharma1911

File: generators/artifacthub/package_test.go [hunk 1]
```diff
@@ -48,7 +48,7 @@ func TestGetChartUrl(t *testing.T) {
 				}
 				byt, _ := json.MarshalIndent(comp, "", "")
 
-				f, err := os.Create(dirName + "/" + comp.Kind + ".json")
+				f, err := os.Create(dirName + "/" + comp.Component.Kind + ".json")
```
Comment (leecalcote) [inline generators/artifacthub/package_test.go:51]: Should we be mandating json?

Comment (leecalcote) [reply generators/artifacthub/package_test.go:51]: Perhaps this is something to create a separate issue about.

File: models/meshmodel/core/policies/data_models.go [hunk 2]
```diff
@@ -17,3 +17,5 @@ type NetworkPolicyRegoResponse struct {
 	ServicePodRelationships        []RelationObject `json:"service_pod_relationships,omitempty"`
 	ServiceDeploymentRelationships []RelationObject `json:"service_deployment_relationships,omitempty"`
 }
+
```
Comment (leecalcote) [inline models/meshmodel/core/policies/data_models.go:20]: @MUzairS15 why do we have relationship-specific structs here?

Comment (leecalcote) [reply models/meshmodel/core/policies/data_models.go:20]: How will the project scale this way?

File: models/meshmodel/core/policies/rego_policy_relationship.go [hunk 3]
```diff
@@ -26,7 +28,10 @@ func NewRegoInstance(policyDir string, regManager *registry.RegistryManager) (*R
 	var store storage.Store
 
 	ctx := context.Background()
-	registeredRelationships, _, _ := regManager.GetEntities(&v1alpha1.RelationshipFilter{})
+	registeredRelationships, _, _, err := regManager.GetEntities(&v1alpha2.RelationshipFilter{})
```
Comment (leecalcote) [inline models/meshmodel/core/policies/rego_policy_relationship.go:31]: This is a concern separate from v1beta1 for Components. 

Relationships v1alpha2 was to have been completely previously, right? Should this go forward independently?

File: models/meshmodel/core/policies/rego_policy_relationship.go [hunk 4]
```diff
@@ -51,7 +56,10 @@ func (r *Rego) RegoPolicyHandler(regoQueryString string, designFile []byte) (int
 	}
 	regoEngine, err := rego.New(
 		rego.Query(regoQueryString),
-		rego.Load([]string{r.policyDir}, nil),
+		rego.Load([]string{r.policyDir}, func(abspath string, info fs.FileInfo, depth int) bool {
```
Comment (leecalcote) [inline models/meshmodel/core/policies/rego_policy_relationship.go:59]: Please add comments here and update Meshery Docs. 

Is this loading Rego policy from each individual model?

File: models/meshmodel/core/v1alpha2/relationship.go [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+}
+func (r RelationshipDefinition) GetID() uuid.UUID {
+	return r.ID
+}
+
+func (r *RelationshipDefinition) GetEntityDetail() string {
+	return fmt.Sprintf("type: %s, definition version: %s, kind: %s, model: %s, version: %s", r.Type(), r.Version, r.Kind, r.Model.Name, r.Model.Version)
+}
+
+func (r *RelationshipDefinition) Create(db *database.Handler, hostID uuid.UUID) (uuid.UUID, error) {
```
Comment (leecalcote) [inline models/meshmodel/core/v1alpha2/relationship.go:47]: Why is there no Swagger / API docs?

File: models/meshmodel/core/v1beta1/application_component.go [hunk 6]
```diff
@@ -1,22 +1,22 @@
...
+// Component is the structure for the core Application Component
 type Component struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	Spec ComponentSpec `json:"spec,omitempty"`
 }
 
-// ComponentSpec is the structure for the core OAM Application Component Spec
+// ComponentSpec is the structure for the core Application Component Spec
```
Comment (leecalcote) [inline models/meshmodel/core/v1beta1/application_component.go:19]: Clean this up, please.

File: models/meshmodel/core/v1beta1/application_component.go [hunk 7]
```diff
@@ -46,7 +46,7 @@ func GetKindFromComponent(comp Component) string {
 	return kind
 }
 
-func GetAnnotationsForWorkload(w v1alpha1.ComponentDefinition) map[string]string {
+func GetAnnotationsForWorkload(w ComponentDefinition) map[string]string {
```
Comment (leecalcote) [inline models/meshmodel/core/v1beta1/application_component.go:49]: What’s a Workload?

File: models/meshmodel/core/v1beta1/component.go [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+
+	"github.com/google/uuid"
+)
+
+type VersionMeta struct {
+	SchemaVersion string `json:"schemaVersion,omitempty" yaml:"schemaVersion"`
+	Version       string `json:"version,omitempty" yaml:"version"`
+}
+
+type TypeMeta struct {
```
Comment (leecalcote) [inline models/meshmodel/core/v1beta1/component.go:20]: Missing comment or swagger docs?

File: models/meshmodel/core/v1beta1/component.go [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+	CUE  ComponentFormat = "CUE"
+)
+
+// Contains information as extracted from the core underlying component eg: Pod's apiVersion, kind and schema
+type ComponentEntity struct {
+	TypeMeta
+	Schema string `json:"schema,omitempty" yaml:"schema"`
+}
+
+// swagger:response ComponentDefinition
```
Comment (leecalcote) [inline models/meshmodel/core/v1beta1/component.go:39]: Is this supposed to be api docs?

**Review by leecalcote — CHANGES_REQUESTED** [review]

File: models/meshmodel/core/policies/rego_policy_relationship.go (hunk 3 above)
Comment (MUzairS15) [reply models/meshmodel/core/policies/rego_policy_relationship.go:31]: Yes, v1alpha2 was completed, but the update was done inplace, i.e. v1alpha1 struct were simply updated in place.

File: models/meshmodel/core/v1alpha2/relationship.go [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+	"gorm.io/gorm/clause"
+)
+
+const SchemaVersion = "core.meshery.io/v1alpha2"
+
+type RelationshipDefinition struct {
+	ID uuid.UUID `json:"id"`
+	v1beta1.VersionMeta
+	Kind string `json:"kind,omitempty" yaml:"kind"`
+	// The property has been named RelationshipType instead of Type to avoid collision from Type() function, which enables support for dynamic type.
```
[the remark this answers is not in the document]
Comment (MUzairS15) [reply models/meshmodel/core/v1alpha2/relationship.go:21]: These comments were added when the relationship defs  were updated to reflect v1alpha2.

**Comment by leecalcote** [comment]
What considerations need to be made for Meshery server upgrades?

File: models/meshmodel/core/v1alpha2/relationship.go (hunk 5 above)
Comment (MUzairS15) [reply models/meshmodel/core/v1alpha2/relationship.go:47]: This is a DB functions, the swagger docs are present for the handles inside `meshery/meshery`

**Comment by MUzairS15** [comment]
> What considerations need to be made for Meshery server upgrades and for Meshery design compatibility?
> 
> Is there a corresponding PR for Meshery extensions (e.g. MeshMap)? @aabidsofi19

Yes, there's a PR in meshery-extensions.

File: models/meshmodel/core/policies/rego_policy_relationship.go (hunk 3 above)
Comment (leecalcote) [reply models/meshmodel/core/policies/rego_policy_relationship.go:31]: If that work was done well and was completed, would we expect that the only change necessary here be that of a swap from “1 “to “2”?

File: generators/artifacthub/package_test.go (hunk 1 above)
Comment (MUzairS15) [reply generators/artifacthub/package_test.go:51]: We don't use this anymore (which was added by Abhi Kumar).
This is leftover to be deleted.

Comment (MUzairS15) [reply generators/artifacthub/package_test.go:51]: deleted

Comment (MUzairS15) [reply generators/artifacthub/package_test.go:51]: comment at wrong place.

File: models/meshmodel/core/policies/data_models.go (hunk 2 above)
Comment (MUzairS15) [reply models/meshmodel/core/policies/data_models.go]: We don't use this anymore (which was added by Abhi Kumar).
This is leftover to be deleted.

Comment (MUzairS15) [reply models/meshmodel/core/policies/data_models.go]: deleted

File: models/meshmodel/core/v1alpha2/relationship.go (hunk 10 above)
[the remark this answers is not in the document]
Comment (leecalcote) [reply models/meshmodel/core/v1alpha2/relationship.go:21]: Oh, good. These are nice to see.

File: models/meshmodel/core/v1beta1/application_component.go (hunk 6 above)
Comment (leecalcote) [reply models/meshmodel/core/v1beta1/application_component.go:20]: File needs renamed?

What is `metav1`? Is there a schema for this?

Comment (MUzairS15) [reply models/meshmodel/core/v1beta1/application_component.go:20]: No, the namings have been inspired from k8s packages, v1 in metav1 eflects the schema version.


These are OAM constructs that is most appropriate to address with v1beta1 version of design schema

**Review by leecalcote — APPROVED** [review]

**Pull request merged**